### PR TITLE
Feature/977 create file endpoint

### PR DESCRIPTION
**Pull request by inkhey**
related to tracim/tracim#977
need https://github.com/algoo/hapic/commit/c79086c29c38b55ee4597640e609f1827b61a130
fix part of tracim/tracim#696
Todo : 
- [X] Add support for parent_id + related test

### CODE REVIEW

File: backend/tracim_backend/models/context_models.py [hunk 1]
```diff
@@ -126,6 +126,21 @@ def __init__(self, loggedin_user_password: str, email: str) -> None:
         self.email = email
 
 
+class SimpleFile(object):
+    """
+    Simple file object
```
Comment (tracim) [inline backend/tracim_backend/models/context_models.py]: useless docstring. Docstring must help understand what this class is usefull for (is it just a container?)... or do not add docstring

File: backend/tracim_backend/models/context_models.py [hunk 2]
```diff
@@ -126,6 +126,21 @@ def __init__(self, loggedin_user_password: str, email: str) -> None:
...
+    """
+    def __init__(self, files):
```
Comment (tracim) [inline backend/tracim_backend/models/context_models.py]: missing typing. what is `files`?

File: backend/tracim_backend/views/core_api/schemas.py [hunk 3]
```diff
@@ -49,11 +50,37 @@
...
+    parent_id = marshmallow.fields.Int(
+        example=2,
+        default=0,
+        validate=Range(min=0, error="Value must be positive or 0"),
+        allow_none=True
+    )
+
+    @post_load
+    def create_set_email_object(self, data):
+        return FileCreation(**data)
```
Comment (tracim) [inline backend/tracim_backend/views/core_api/schemas.py]: I really do not understand the naming strategy and the associated logic: FieCreation is a container with a unique property parent_id, and it is set through the call to a `create_set_email_object()` method

File: backend/tracim_backend/views/contents_api/file_controller.py [hunk 4]
```diff
@@ -55,11 +62,67 @@ class FileController(Controller):
...
+        """
+        app_config = request.registry.settings['CFG']
+        api = ContentApi(
+            show_archived=True,
+            show_deleted=True,
+            current_user=request.current_user,
+            session=request.dbsession,
+            config=app_config,
+        )
+        file = hapic_data.files.files
```
Comment (tracim) [inline backend/tracim_backend/views/contents_api/file_controller.py]: do not name a local var `file`. Good habit is to name it f or _file... by the way, why do you name it file while in hapic_data its files (plural form)?

File: backend/tracim_backend/views/contents_api/file_controller.py [hunk 5]
```diff
@@ -55,11 +62,67 @@ class FileController(Controller):
...
+        parent_id = hapic_data.forms.parent_id
+        api = ContentApi(
+            current_user=request.current_user,
+            session=request.dbsession,
+            config=app_config
+        )
+
+        parent = None
```
Comment (tracim) [inline backend/tracim_backend/views/contents_api/file_controller.py]: typing is missing

File: backend/tracim_backend/views/core_api/schemas.py (hunk 3 above)
Comment (buxx) [reply backend/tracim_backend/views/core_api/schemas.py]: @inkhey Some information about this remark ?

Comment (inkhey) [reply backend/tracim_backend/views/core_api/schemas.py]: @buxx create_set_email_object was renamed in order to make  sense.

Comment (buxx) [reply backend/tracim_backend/views/core_api/schemas.py]: okay

**Pull request merged**